exercises/custom-set/example.rs

```diff
@@ -1,149 +1,100 @@
-use std::iter::{IntoIterator, FromIterator};
-use std::slice;
-use std::option::Option;
-use std::vec::Vec;
-
 #[derive(Debug)]
 pub struct CustomSet<T> {
-    elements: Vec<T>
+    collection: Vec<T>,
 }
 
-/// Create an empty set.
-impl <T> CustomSet<T> where T: Eq + Clone {
-    pub fn new() -> CustomSet<T> {
-        CustomSet { elements: Vec::new() }
+impl<T: Ord + Clone> PartialEq for CustomSet<T> {
+    fn eq(&self, other: &Self) -> bool {
+        self.collection.iter().all(|x| other.contains(&x)) &&
+        other.collection.iter().all(|x| self.contains(&x))
     }
+}
 
-    pub fn len(&self) -> usize {
-        self.elements.len()
+impl<T: PartialEq + Ord + Clone> CustomSet<T> {
```

Comment:
out of curiosity: since https://doc.rust-lang.org/std/cmp/trait.Ord.html tells me `Ord` depends on `PartialEq`, does having `PartialEq` here make a difference?